consensus/hotstuff/timeoutaggregator/timeout_aggregator.go

```diff
@@ -198,12 +198,22 @@ func (t *TimeoutAggregator) PruneUpToView(lowestRetainedView uint64) {
 	t.collectors.PruneUpToView(lowestRetainedView)
 }
 
-// OnEnteringView implements the `OnEnteringView` callback from the `hotstuff.Consumer`.
+// OnQcTriggeredViewChange implements the `OnQcTriggeredViewChange` callback from the `hotstuff.Consumer`.
 // We notify the enteringViewProcessingLoop worker, which then prunes up to the active view.
 // CAUTION: the input to this callback is treated as trusted; precautions should be taken that messages
 // from external nodes cannot be considered as inputs to this function
-func (t *TimeoutAggregator) OnEnteringView(viewNumber uint64, _ flow.Identifier) {
-	if t.lowestRetainedView.Set(viewNumber) {
+func (t *TimeoutAggregator) OnQcTriggeredViewChange(_ *flow.QuorumCertificate, newView uint64) {
+	if t.lowestRetainedView.Set(newView) {
+		t.enteringViewNotifier.Notify()
+	}
+}
+
+// OnTcTriggeredViewChange implements the `OnTcTriggeredViewChange` callback from the `hotstuff.Consumer`.
+// We notify the enteringViewProcessingLoop worker, which then prunes up to the active view.
+// CAUTION: the input to this callback is treated as trusted; precautions should be taken that messages
+// from external nodes cannot be considered as inputs to this function
+func (t *TimeoutAggregator) OnTcTriggeredViewChange(_ *flow.TimeoutCertificate, newView uint64) {
```

Comment:
In this structure we are interested in receiving `ViewChange` events, but it requires overloading two functions, which is error-prone. Should we add a specific callback for it? The drawback that active pacemaker will be calling 3 events:
1. `OnQcTriggeredViewChange` or `OnTcTriggeredViewChange`
2. `OnStartingTimeout`
3. `OnViewChange`

Reply:
I think a generic `OnViewChange` event is sufficiently useful to justify itself. 

I'd be OK to leave this as-is if it's likely to be the only usage, but I feel like generic view change is something that could be useful in a lot of places.

Reply:
Agree: https://github.com/onflow/flow-go/pull/3571/commits/aac1fa6f990797174af7185a7391cfc3c35f834a